sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -530,7 +531,10 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     s"$nodeName ${argString(maxFields)}".trim
   }
 
-  /** ONE line description of this node with more information */
+  def simpleString(
+     planToOperatorID: mutable.LinkedHashMap[QueryPlan[_], Int]): String = s"$nodeName".trim
```

Comment:
It's weird to see `planToOperatorID` parameter in TreeNode.

Reply:
@cloud-fan I agree. Actually i tried very hard make place the new methods in QueryPlan and keep TreeNode untouched. But couldn't do it as i was operating with a premise to not modify the plan in anyway. But your subsequent comment to use tagging to store the operatorid temporarily in the plan may help. Let me think about this a bit.